lib/twap/util/generate_order.js

```diff
@@ -20,13 +22,22 @@ const { DUST } = Config
 const generateOrder = (state = {}, price) => {
   const { args = {}, remainingAmount } = state
   const {
-    sliceAmount, orderType, symbol, amount, lev, _margin, _futures
+    sliceAmount, orderType, symbol, amount, lev, _margin, _futures, amountDistortion
   } = args
+  let orderAmount
 
+  if (_isFinite(amountDistortion)) {
+    const plusOrMinus = Math.random() > 0.5 ? 1 : -1
+    const randomPercentage = ((Math.random() * amountDistortion * plusOrMinus) + 1) // random percentage from 1 to amountDistortion
```

Comment:
just to double check, lets say i choose a 2% distortion.

i have an initial value of 100 - so the submtited values are withign the range of 98 - 102, correct?

can you describe and test if everything works with the overall order amount?

e.g. i have an overall amount of 100, distortion 2%, with 5 slices...

does the algo then could overuse amounts from adding up additional distortions?

Reply:
the submitted orders never exceeds the total amount entered by the user despite the additional distortions.. in every interval tick, if tradeBeyondEnd is enabled, the algo ensures that the next order would not push us over the total amount and doesnt submit new orders; and when tradeBeyondEnd is disabled, the remaining amount is sent for the new order when the next order pushes over the total amount 
i will write the test cases covering these possible cases